src/coreclr/tools/Common/TypeSystem/Common/MethodDesc.cs

```diff
@@ -284,6 +294,21 @@ private bool Equals(MethodSignature otherSignature, bool allowCovariantReturn)
             }
 
             return false;
+
+            static bool CompareTypeHelper(TypeDesc type1, TypeDesc type2, bool allowEquivalence, StackOverflowProtect visited)
```

Comment:
Naming nit - both my old C++ background and Compare-like C# APIs typically return plus / minus / zero result of the comparison whereas this feels more like something in the vein of <code>IsTypeEqualHelper</code>. Not a big deal but I had to fight this mental model half a dozen times while reviewing this change.